### PR TITLE
feat(backend): serve jwks.json from client payment pointer

### PR DESCRIPTION
**Pull request by wilsonianb**
## Changes proposed in this pull request

- Configure the RS's client payment pointer and serve its jwk (derived from the configured private key)

## Context

- progress toward #583 

## Checklist

- [ ] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/open-payments/src/client/index.ts [hunk 1]
```diff
@@ -125,7 +126,8 @@ export const createAuthenticatedClient = async (
     grant: createGrantRoutes({
       axiosInstance,
       openApi: authServerOpenApi,
-      logger
+      logger,
+      client: args.client
```
Comment (wilsonianb) [inline packages/open-payments/src/client/index.ts:130]: I was going to have `createAuthenticatedClient` query `${client}/jwks.json` to validate that the key with the specified `keyid` is served there, but `backend` calls `createAuthenticatedClient` long before its routes are set up.

Comment (mkurapov) [reply packages/open-payments/src/client/index.ts:130]: Yeah I was thinking you'd possibly get some kind of race condition type behaviour - could add some kind of retry mechanism but gets too complicated for what it's worth. 

 Is there a more straightforward name than `client`? Even just `paymentPointerUrl`? or `clientPaymentPointer`?

File: packages/backend/src/open_payments/payment_pointer/middleware.ts [hunk 2]
```diff
@@ -5,16 +5,18 @@ export function createPaymentPointerMiddleware() {
...
-      'paymentPointerService'
-    )
-    const paymentPointer = await paymentPointerService.getByUrl(
-      `https://${ctx.request.host}/${ctx.params.paymentPointerPath}`
-    )
-    if (!paymentPointer) {
-      ctx.throw(404)
+    const url = `https://${ctx.request.host}/${ctx.params.paymentPointerPath}`
+    const config = await ctx.container.use('config')
+    if (url !== config.paymentPointerUrl) {
```
Comment (mkurapov) [inline packages/backend/src/open_payments/payment_pointer/middleware.ts:10]: In which situations will this equality happen?

Comment (wilsonianb) [reply packages/backend/src/open_payments/payment_pointer/middleware.ts:10]: When this is an Open Payments resource request for the RS's own payment pointer, which will now/soon be happening when a separate AS is querying RS's client payment pointer `/jwks.json`

File: packages/backend/src/paymentPointerKey/routes.ts [hunk 3]
```diff
@@ -24,15 +38,22 @@ export async function getKeysByPaymentPointerId(
...
+    )
 
-  ctx.body = {
-    keys: keys.map((key) => key.jwk)
+    ctx.body = {
+      keys: keys.map((key) => key.jwk)
+    }
+  } else if (
+    deps.config.paymentPointerUrl ===
+    `https://${ctx.request.host}/${ctx.params.paymentPointerPath}`
```
Comment (mkurapov) [inline packages/backend/src/paymentPointerKey/routes.ts:51]: Should this be made into a function we can use across this package? Any function name will also help with the self-commenting of the code IMO

File: packages/open-payments/src/client/index.ts [hunk 4]
```diff
@@ -93,6 +93,7 @@ export interface CreateAuthenticatedClientArgs
   extends CreateUnauthenticatedClientArgs {
```
Comment (mkurapov) [inline packages/open-payments/src/client/index.ts:93]: Kind of outside the scope of this PR, but I'm a bit confused why TS doesn't complain about
```ts
const axiosInstance = createAxiosInstance({
    privateKey: args.privateKey,
    keyId: args.keyId,
    requestTimeoutMs:
      args?.requestTimeoutMs ?? config.DEFAULT_REQUEST_TIMEOUT_MS
}
```
when privateKey and keyId are optional in `createUnauthenticatedClient`.

**Review by mkurapov — DISMISSED** [review]
Few comments, otherwise LGTM :+1:

File: packages/backend/src/paymentPointerKey/routes.ts (hunk 3 above)
Comment (wilsonianb) [reply packages/backend/src/paymentPointerKey/routes.ts:51]: I'd like to be able to do this via a `paymentPointerUrl` getter on `AppContext`(?), but I'm struggling to pull it off.

File: packages/open-payments/src/client/index.ts (hunk 4 above)
Comment (wilsonianb) [reply packages/open-payments/src/client/index.ts:93]: I think this due to
https://github.com/interledger/rafiki/blob/2f17be36bb705e3afbe449d577d380e79e876646/tsconfig.build.json#L28
which was toggled in
- https://github.com/interledger/rafiki/pull/758

@matdehaast @traviscrist Do either of you remember why we disabled strict type checking?

Comment (wilsonianb) [reply packages/open-payments/src/client/index.ts:93]: >If you enable it how many issues does it find?

tbd
- https://github.com/interledger/rafiki/pull/768

**Pull request merged**